Failure to build on mac

### Company or project name

_No response_

### Operating system

Mac

### CMake version

cmake version 4.0.1

### Ninja version

1.12.1

### Compiler name and version

Clang:
Homebrew clang version 20.1.3
Target: arm64-apple-darwin24.4.0
Thread model: posix
InstalledDir: /opt/homebrew/Cellar/llvm/20.1.3/bin
Configuration file: /opt/homebrew/etc/clang/arm64-apple-darwin24.cfg

Clang++:
Homebrew clang version 20.1.3
Target: arm64-apple-darwin24.4.0
Thread model: posix
InstalledDir: /opt/homebrew/Cellar/llvm/20.1.3/bin
Configuration file: /opt/homebrew/etc/clang/arm64-apple-darwin24.cfg

```
➜  ClickHouse git:(master) ✗ which clang
/opt/homebrew/opt/llvm/bin/clang
➜  ClickHouse git:(master) ✗ which clang++
/opt/homebrew/opt/llvm/bin/clang++
```

### Full cmake and/or ninja output

Steps for repro
- clone from git, pull master
- pull all submodules as mentioned in the docs
- follow https://clickhouse.com/docs/development/build-osx
- `cmake --build build` fails

Previous issue that seems similar: https://github.com/ClickHouse/ClickHouse/issues/8085

Output:

```
➜  ClickHouse git:(master) ✗ cmake --build build                        
[0/2] Re-checking globbed directories...
[0/9] cd /Users/elisylvia-lourde/git/ClickHouse/rust/workspace/prq...ke/config.toml --offline --locked -- -Cdefault-linker-libraries=no
    Finished `release` profile [optimized] target(s) in 0.10s
[1/9] cd /Users/elisylvia-lourde/git/ClickHouse/rust/workspace/ski...ke/config.toml --offline --locked -- -Cdefault-linker-libraries=no
    Finished `release` profile [optimized] target(s) in 0.10s
[5/7] Linking CXX executable src/Common/mysqlxx/tests/mysqlxx_pool_test
FAILED: src/Common/mysqlxx/tests/mysqlxx_pool_test 
: && /opt/homebrew/opt/llvm/bin/clang++ -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8.2-a+simd+crypto+dotprod+ssbs+rcpc+bf16 -ffile-prefix-map=/Users/elisylvia-lourde/git/ClickHouse=. -falign-functions=32 -ffp-contract=off  -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fPIC -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wvla-cxx-extension -Wno-return-type-c-linkage -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-unsafe-buffer-usage -Wno-switch-default -O2 -g -DNDEBUG -O3 -g  -arch arm64 -mmacosx-version-min=10.15 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-no_warn_duplicate_libraries --ld-path=/usr/bin/ld -Wl,-U,_inside_main -u_zone_register src/CMakeFiles/clickhouse_malloc.dir/Common/malloc.cpp.o src/Common/mysqlxx/tests/CMakeFiles/mysqlxx_pool_test.dir/mysqlxx_pool_test.cpp.o -o src/Common/mysqlxx/tests/mysqlxx_pool_test  src/Common/mysqlxx/libmysqlxx.a  src/libdbms.a  src/Common/Config/libclickhouse_common_config.a  src/Loggers/libloggers_no_text_log.a  contrib/yaml-cpp-cmake/lib_yaml_cpp.a  contrib/cppkafka-cmake/lib_cppkafka.a  contrib/librdkafka-cmake/lib_rdkafka.a  contrib/nats-io-cmake/lib_nats_io.a  contrib/cyrus-sasl-cmake/lib_sasl2.a  src/Common/ZooKeeper/libclickhouse_common_zookeeper_base.a  src/Common/ZooKeeper/libclickhouse_common_zookeeper.a  src/Dictionaries/Embedded/libclickhouse_dictionaries_embedded.a  src/Common/mysqlxx/libmysqlxx.a  contrib/mariadb-connector-c-cmake/lib_mariadbclient.a  -liconv  src/libclickhouse_compression.a  contrib/llvm-project/llvm/lib/libLLVMSupport.a  contrib/llvm-project/llvm/lib/libLLVMDemangle.a  contrib/croaring-cmake/lib_roaring.a  contrib/nuraft-cmake/lib_nuraft.a  contrib/boost-cmake/lib_boost_coroutine.a  src/Parsers/libclickhouse_parsers.a  src/Access/Common/libclickhouse_common_access.a  rust/workspace/lib_ch_rust_prql.a  base/poco/Redis/lib_poco_redis.a  contrib/mongo-cxx-driver-cmake/lib_mongocxx.a  contrib/mongo-c-driver-cmake/lib_libmongoc.a  contrib/mongo-cxx-driver-cmake/lib_bsoncxx.a  contrib/mongo-c-driver-cmake/lib_libbson.a  contrib/icu-cmake/lib_icui18n.a  contrib/icu-cmake/lib_icuuc.a  contrib/icu-cmake/lib_icudata.a  contrib/capnproto-cmake/lib_capnpc.a  contrib/capnproto-cmake/lib_capnp.a  contrib/capnproto-cmake/lib_kj.a  contrib/arrow-cmake/lib_parquet.a  contrib/arrow-cmake/lib_arrow.a  contrib/boost-cmake/lib_boost_filesystem.a  contrib/arrow-cmake/lib_flatbuffers.a  contrib/arrow-cmake/lib_orc.a  contrib/avro-cmake/lib_avrocpp.a  contrib/boost-cmake/lib_boost_iostreams.a  contrib/openldap-cmake/lib_ldap.a  contrib/openldap-cmake/lib_lber.a  src/Server/grpc_protos/libclickhouse_grpc_protos.a  contrib/grpc-cmake/libgrpc++.a  contrib/grpc-cmake/libgrpc.a  contrib/grpc-cmake/libupb_json_lib.a  contrib/grpc-cmake/libupb_textformat_lib.a  contrib/grpc-cmake/libupb_collections_lib.a  contrib/grpc-cmake/libupb.a  contrib/grpc-cmake/libutf8_range_lib.a  contrib/grpc-cmake/libgpr.a  contrib/abseil-cpp-cmake/libabsl_random_distributions.a  contrib/abseil-cpp-cmake/libabsl_random_seed_sequences.a  contrib/abseil-cpp-cmake/libabsl_random_internal_pool_urbg.a  contrib/abseil-cpp-cmake/libabsl_random_internal_randen.a  contrib/abseil-cpp-cmake/libabsl_random_internal_randen_hwaes.a  contrib/abseil-cpp-cmake/libabsl_random_internal_randen_hwaes_impl.a  contrib/abseil-cpp-cmake/libabsl_random_internal_randen_slow.a  contrib/abseil-cpp-cmake/libabsl_random_internal_platform.a  contrib/abseil-cpp-cmake/libabsl_random_internal_seed_material.a  contrib/abseil-cpp-cmake/libabsl_random_seed_gen_exception.a  contrib/grpc-cmake/libaddress_sorting.a  -framework CoreFoundation  contrib/libhdfs3-cmake/lib_hdfs3.a  contrib/libgsasl-cmake/lib_gsasl.a  contrib/krb5-cmake/lib_krb5.a  contrib/hive-metastore-cmake/lib_hivemetastore.a  contrib/thrift-cmake/lib_thrift.a  contrib/s2geometry-cmake/lib_s2.a  contrib/vectorscan-cmake/lib_vectorscan.a  contrib/amqpcpp-cmake/lib_amqp-cpp.a  contrib/libuv-cmake/lib_uv.a  contrib/sqlite-cmake/lib_sqlite.a  contrib/rocksdb-cmake/lib_rocksdb.a  contrib/libpqxx-cmake/lib_libpqxx.a  contrib/postgres-cmake/lib_libpq.a  contrib/libstemmer-c-cmake/lib_stemmer.a  contrib/wordnet-blast-cmake/lib_wnb.a  contrib/boost-cmake/lib_boost_graph.a  contrib/boost-cmake/lib_boost_regex.a  contrib/lemmagen-c-cmake/lib_lemmagen.a  contrib/ulid-c-cmake/lib_ulid.a  contrib/consistent-hashing/lib_consistent_hashing.a  contrib/SimSIMD-cmake/lib_simsimd.a  contrib/prometheus-protobufs-cmake/lib_prometheus_protobufs.a  rust/workspace/lib_ch_rust_skim_rust.a  contrib/sha3iuf-cmake/lib_sha3iuf.a  src/libclickhouse_common_io.a  base/poco/JSON/lib_poco_json.a  base/base/libcommon.a  base/poco/XML/lib_poco_xml.a  base/poco/Foundation/lib_poco_foundation.a  base/poco/Net/lib_poco_net.a  base/poco/NetSSL_OpenSSL/lib_poco_net_ssl.a  base/poco/Util/lib_poco_util.a  src/libclickhouse_common_io.a  base/poco/JSON/lib_poco_json.a  base/base/libcommon.a  base/poco/XML/lib_poco_xml.a  base/poco/Foundation/lib_poco_foundation.a  base/poco/Net/lib_poco_net.a  base/poco/NetSSL_OpenSSL/lib_poco_net_ssl.a  base/poco/Util/lib_poco_util.a  contrib/jemalloc-cmake/lib_jemalloc.a  contrib/boost-cmake/lib_boost_program_options.a  contrib/google-protobuf-cmake/lib_libprotobuf.a  -lpthread  contrib/google-protobuf-cmake/libutf8_validity.a  contrib/abseil-cpp-cmake/libabsl_leak_check.a  contrib/abseil-cpp-cmake/libabsl_die_if_null.a  contrib/abseil-cpp-cmake/libabsl_log_initialize.a  contrib/abseil-cpp-cmake/libabsl_statusor.a  contrib/abseil-cpp-cmake/libabsl_status.a  contrib/azure-cmake/lib_azure_sdk.a  contrib/curl-cmake/lib_curl.a  contrib/libxml2-cmake/lib_libxml2.a  contrib/boost-cmake/lib_boost_context.a  contrib/simdjson-cmake/lib_simdjson.a  base/widechar_width/libwidechar_width.a  contrib/dragonbox-cmake/lib_dragonbox_to_chars.a  contrib/re2-cmake/lib_re2.a  contrib/abseil-cpp-cmake/libabsl_flags_internal.a  contrib/abseil-cpp-cmake/libabsl_flags_marshalling.a  contrib/abseil-cpp-cmake/libabsl_flags_reflection.a  contrib/abseil-cpp-cmake/libabsl_flags_config.a  contrib/abseil-cpp-cmake/libabsl_flags_program_name.a  contrib/abseil-cpp-cmake/libabsl_flags_private_handle_accessor.a  contrib/abseil-cpp-cmake/libabsl_flags_commandlineflag.a  contrib/abseil-cpp-cmake/libabsl_flags_commandlineflag_internal.a  contrib/abseil-cpp-cmake/libabsl_cord.a  contrib/abseil-cpp-cmake/libabsl_cordz_info.a  contrib/abseil-cpp-cmake/libabsl_cord_internal.a  contrib/abseil-cpp-cmake/libabsl_cordz_functions.a  contrib/abseil-cpp-cmake/libabsl_cordz_handle.a  contrib/abseil-cpp-cmake/libabsl_crc_cord_state.a  contrib/abseil-cpp-cmake/libabsl_crc32c.a  contrib/abseil-cpp-cmake/libabsl_crc_internal.a  contrib/abseil-cpp-cmake/libabsl_crc_cpu_detect.a  contrib/abseil-cpp-cmake/libabsl_raw_hash_set.a  contrib/abseil-cpp-cmake/libabsl_hashtablez_sampler.a  contrib/abseil-cpp-cmake/libabsl_exponential_biased.a  contrib/abseil-cpp-cmake/libabsl_log_internal_check_op.a  contrib/abseil-cpp-cmake/libabsl_log_internal_conditions.a  contrib/abseil-cpp-cmake/libabsl_log_internal_message.a  contrib/abseil-cpp-cmake/libabsl_strerror.a  contrib/abseil-cpp-cmake/libabsl_log_internal_nullguard.a  contrib/abseil-cpp-cmake/libabsl_examine_stack.a  contrib/abseil-cpp-cmake/libabsl_log_internal_format.a  contrib/abseil-cpp-cmake/libabsl_str_format_internal.a  contrib/abseil-cpp-cmake/libabsl_log_internal_proto.a  contrib/abseil-cpp-cmake/libabsl_log_internal_log_sink_set.a  contrib/abseil-cpp-cmake/libabsl_log_globals.a  contrib/abseil-cpp-cmake/libabsl_hash.a  contrib/abseil-cpp-cmake/libabsl_bad_variant_access.a  contrib/abseil-cpp-cmake/libabsl_city.a  contrib/abseil-cpp-cmake/libabsl_low_level_hash.a  contrib/abseil-cpp-cmake/libabsl_log_internal_globals.a  contrib/abseil-cpp-cmake/libabsl_log_sink.a  contrib/abseil-cpp-cmake/libabsl_log_entry.a  contrib/abseil-cpp-cmake/libabsl_vlog_config_internal.a  contrib/abseil-cpp-cmake/libabsl_log_internal_fnmatch.a  contrib/abseil-cpp-cmake/libabsl_bad_optional_access.a  contrib/abseil-cpp-cmake/libabsl_synchronization.a  contrib/abseil-cpp-cmake/libabsl_stacktrace.a  contrib/abseil-cpp-cmake/libabsl_symbolize.a  contrib/abseil-cpp-cmake/libabsl_debugging_internal.a  contrib/abseil-cpp-cmake/libabsl_demangle_internal.a  contrib/abseil-cpp-cmake/libabsl_graphcycles_internal.a  contrib/abseil-cpp-cmake/libabsl_kernel_timeout_internal.a  contrib/abseil-cpp-cmake/libabsl_time.a  contrib/abseil-cpp-cmake/libabsl_strings.a  contrib/abseil-cpp-cmake/libabsl_int128.a  contrib/abseil-cpp-cmake/libabsl_strings_internal.a  contrib/abseil-cpp-cmake/libabsl_string_view.a  contrib/abseil-cpp-cmake/libabsl_throw_delegate.a  contrib/abseil-cpp-cmake/libabsl_civil_time.a  contrib/abseil-cpp-cmake/libabsl_time_zone.a  -Wl,-framework,CoreFoundation  contrib/abseil-cpp-cmake/libabsl_malloc_internal.a  contrib/abseil-cpp-cmake/libabsl_base.a  contrib/abseil-cpp-cmake/libabsl_spinlock_wait.a  contrib/abseil-cpp-cmake/libabsl_raw_logging_internal.a  contrib/abseil-cpp-cmake/libabsl_log_severity.a  contrib/libfiu-cmake/lib_libfiu.a  contrib/libssh-cmake/lib_ssh.a  contrib/c-ares-cmake/lib_c-ares.a  contrib/aws-cmake/lib_aws.a  contrib/brotli-cmake/lib_brotli.a  contrib/snappy-cmake/lib_snappy.a  contrib/libbcrypt-cmake/lib_bcrypt.a  contrib/libarchive-cmake/lib_libarchive.a  contrib/minizip-ng-cmake/lib_minizip.a  contrib/zstd-cmake/lib_zstd.a  contrib/xz-cmake/lib_liblzma.a  contrib/bzip2-cmake/lib_bzip2.a  base/poco/JSON/lib_poco_json_pdjson.a  contrib/boost-cmake/lib_boost_system.a  contrib/cityhash102/lib_cityhash.a  contrib/replxx-cmake/lib_replxx.a  contrib/cctz-cmake/lib_cctz.a  contrib/cctz-cmake/libtzdata.a  contrib/fmtlib-cmake/lib_fmt.a  base/poco/XML/lib_poco_xml_expat.a  contrib/zlib-ng-cmake/lib_zlib.a  contrib/lz4-cmake/lib_lz4.a  contrib/double-conversion-cmake/lib_double-conversion.a  base/poco/Foundation/lib_poco_foundation_pcre.a  contrib/openssl-cmake/libssl.a  contrib/openssl-cmake/libcrypto.a  contrib/libcxx-cmake/libcxx.a  contrib/libcxxabi-cmake/libcxxabi.a  contrib/libunwind-cmake/libunwind.a  -nodefaultlibs  -lc -lm -lpthread -ldl && :
Undefined symbols for architecture arm64:
  "DB::KeyValuePairExtractorBuilder::withItemDelimiters(std::__1::vector<char, std::__1::allocator<char>>)", referenced from:
      DB::VirtualColumnUtils::makeExtractor() in libdbms.a[935](VirtualColumnUtils.cpp.o)
  "DB::KeyValuePairExtractorBuilder::withKeyValueDelimiter(char)", referenced from:
      DB::VirtualColumnUtils::makeExtractor() in libdbms.a[935](VirtualColumnUtils.cpp.o)
  "DB::extractKV::ConfigurationFactory::createWithoutEscaping(char, char, std::__1::vector<char, std::__1::allocator<char>>)", referenced from:
      DB::KeyValuePairExtractorBuilder::buildWithReferenceMap() const in libdbms.a[935](VirtualColumnUtils.cpp.o)
ld: symbol(s) not found for architecture arm64
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
[6/7] Stripping clickhouse binary
ninja: build stopped: subcommand failed.
```

Note that I re-ran this command, so previous progress was saved in cache

--- a/src/Common/mysqlxx/tests/CMakeLists.txt
+++ b/src/Common/mysqlxx/tests/CMakeLists.txt
@@ -1,2 +1,2 @@
 clickhouse_add_executable (mysqlxx_pool_test mysqlxx_pool_test.cpp)
-target_link_libraries (mysqlxx_pool_test PRIVATE mysqlxx dbms clickhouse_common_config loggers_no_text_log)
+target_link_libraries (mysqlxx_pool_test PRIVATE mysqlxx dbms clickhouse_common_config loggers_no_text_log clickhouse_functions_extractkeyvaluepairs)

